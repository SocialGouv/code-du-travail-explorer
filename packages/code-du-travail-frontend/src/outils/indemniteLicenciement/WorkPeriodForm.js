import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const WorkPeriodForm = React.forwardRef(function WorkPeriodFormRef(
  {
    period: { uid, type, duree = "", salaire = "", isValid },
    onSubmit,
    onChange
  },
  ref
) {
  return (
    <form onSubmit={onSubmit}>
      <Wrapper>
        <InlineLabel>
          <SpanLabel>type</SpanLabel>
          <SelectInput
            ref={ref}
            name="type"
            onChange={onChange}
            onBlur={onChange}
            value={type}
          >
            <option value="temps-partiel">Temps partiel</option>
            <option value="temps-plein">Temps plein</option>
          </SelectInput>
        </InlineLabel>
        <InlineLabel>
          <SpanLabel>durée (en mois)</SpanLabel>
          <DureeInput
            onChange={onChange}
            type="number"
            min="0"
            name="duree"
            placeholder="8 mois"
            value={duree}
          />
        </InlineLabel>
        <InlineLabel>
          <SpanLabel>salaire mensuel</SpanLabel>
          <SalaireInput
            onChange={onChange}
            type="number"
            min="0"
            name="salaire"
            placeholder="€"
            value={salaire}
          />
        </InlineLabel>
        {uid ? (
          <button className="btn" disabled={!isValid}>
            Modifier la période
          </button>
        ) : (
          <button className="btn" disabled={!isValid}>
            Ajouter la période
          </button>
        )}
      </Wrapper>
    </form>
  );
});

WorkPeriodForm.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  period: PropTypes.shape({
    uid: PropTypes.string,
    type: PropTypes.string,
    duree: PropTypes.string,
    salaire: PropTypes.string
  }).isRequired
};

export { WorkPeriodForm };

const Wrapper = styled.div`
  display: flex;
  justify-content: space-between;
  margin-top: 2em;
`;
const SelectInput = styled.select`
  width: 14rem;
`;

const DureeInput = styled.input`
  width: 8.5rem;
`;
const SalaireInput = styled.input`
  width: 8.5rem;
`;
const InlineLabel = styled.label`
  position: relative;
`;
const SpanLabel = styled.span`
  display: block;
  position: absolute;
  top: -1.5rem;
`;
