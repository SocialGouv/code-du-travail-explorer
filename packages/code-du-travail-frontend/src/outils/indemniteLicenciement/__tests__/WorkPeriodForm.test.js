import React from "react";
import { render, fireEvent } from "react-testing-library";
import { WorkPeriodForm } from "../WorkPeriodForm";

describe("<WorkPeriodForm />", () => {
  it("should render single work period", () => {
    const onChange = jest.fn();
    const onSubmit = jest.fn();
    const period = {
      type: "temps-plein",
      duree: "10",
      salaire: "1000"
    };
    const ref = React.createRef();
    const { container } = render(
      <WorkPeriodForm
        period={period}
        onChange={onChange}
        onSubmit={onSubmit}
        ref={ref}
      />
    );
    expect(container).toMatchSnapshot();
  });
  it("should call onChange", () => {
    const props = {
      onChange: jest.fn(),
      onSubmit: jest.fn(),
      period: {
        type: "temps-plein",
        duree: "10",
        salaire: "1000"
      }
    };
    const ref = React.createRef();
    const { getByLabelText } = render(<WorkPeriodForm {...props} ref={ref} />);

    const salaireInput = getByLabelText(/salaire/i);
    fireEvent.change(salaireInput, { target: { value: 1500 } });
    expect(props.onChange).toHaveBeenCalled();
  });

  it("should call onSubmit", () => {
    const props = {
      onChange: jest.fn(),
      // HACK(lionelb): needed to avoid jsdom error about submit
      // @see https://github.com/jsdom/jsdom/issues/1937
      onSubmit: jest.fn().mockImplementation(event => event.preventDefault()),
      period: {
        type: "temps-plein",
        duree: "10",
        salaire: "1000"
      }
    };
    const ref = React.createRef();
    const { getByText } = render(<WorkPeriodForm {...props} ref={ref} />);

    const addButton = getByText(/ajouter la p√©riode/i);
    addButton.click();
    expect(props.onSubmit).toHaveBeenCalled();
  });
});
