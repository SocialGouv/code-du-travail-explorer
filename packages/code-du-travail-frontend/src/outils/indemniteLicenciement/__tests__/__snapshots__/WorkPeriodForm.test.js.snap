// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WorkPeriodForm /> should render single work period 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 2em;
}

.c3 {
  width: 14rem;
}

.c4 {
  width: 8.5rem;
}

.c5 {
  width: 8.5rem;
}

.c1 {
  position: relative;
}

.c2 {
  display: block;
  position: absolute;
  top: -1.5rem;
}

<div>
  <form>
    <div
      class="c0"
    >
      <label
        class="c1"
      >
        <span
          class="c2"
        >
          type
        </span>
        <select
          class="c3"
          name="type"
        >
          <option
            value="temps-partiel"
          >
            Temps partiel
          </option>
          <option
            value="temps-plein"
          >
            Temps plein
          </option>
        </select>
      </label>
      <label
        class="c1"
      >
        <span
          class="c2"
        >
          durée (en mois)
        </span>
        <input
          class="c4"
          min="0"
          name="duree"
          placeholder="8 mois"
          type="number"
          value="10"
        />
      </label>
      <label
        class="c1"
      >
        <span
          class="c2"
        >
          salaire mensuel
        </span>
        <input
          class="c5"
          min="0"
          name="salaire"
          placeholder="€"
          type="number"
          value="1000"
        />
      </label>
      <button
        class="btn"
        disabled=""
      >
        Ajouter la période
      </button>
    </div>
  </form>
</div>
`;
